src/views.js

```diff
@@ -354,7 +358,19 @@ function getPreviewForFile(path, thumbnailFactory) {
         let thumbnailPath = info.get_attribute_byte_string(Gio.FILE_ATTRIBUTE_THUMBNAIL_PATH);
         let thumbnail = null;
 
-        if (thumbnailPath && GLib.file_test(thumbnailPath, GLib.FileTest.EXISTS)) {
+        // We need to check that both the thumbnail exists and that the
+        // mtime of the thumbnail is greater than the mtime of the original
+        // file
+        let mtimeThumbnail = (Gio.File.new_for_path(thumbnailPath)).query_info([
+            Gio.FILE_ATTRIBUTE_TIME_MODIFIED
+        ].join(','), Gio.FileQueryInfoFlags.NONE, null).get_modification_time();
+
+        let mTimeUsecs = timevalToUsecs(mtime);
+        let mTimeThumbnailUsecs = timevalToUsecs(mtimeThumbnail);
```

Comment:
Instead of all of this, you should just check the `G_FILE_ATTRIBUTE_THUMBNAIL_IS_VALID` attribute, which does the same (and possibly more checks) internally